src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -161,71 +161,24 @@ namespace OpenMS
       double best_left, double best_right, double peak_height, double & background, double & avg_noise_level)
   {
-    // determine (in the chromatogram) the intensity at the left / right border
-    double intensity_left = 0.0;
-    double rt_apex = 0.0;
-    double intensity_right = 0.0;
-    
-    // calculate the average noise level
-    for (MSChromatogram::const_iterator it = std::next(chromatogram.begin()); it != chromatogram.end(); ++it)
+    double peak_apex_RT(0.0);
+    double max_height = 0.0;
+    for (MSChromatogram::ConstIterator it = chromatogram.RTBegin(best_left); it != chromatogram.RTEnd(best_right); ++it)
     {
-      MSChromatogram::const_iterator it_prev = it;
-      --it_prev; //previous point
-
-      if (it->getMZ() >= best_left && it_prev->getMZ() < best_left)
-      {
-        intensity_left = it->getIntensity();
-      }
-      else if (it->getIntensity() >= peak_height && it_prev->getIntensity() < peak_height)
-      {
-        rt_apex = it->getMZ();
-      }
-      else if (it->getMZ() >= best_right && it_prev->getMZ() < best_right)
+      if (max_height < it->getIntensity())
       {
-        intensity_right = it->getIntensity();
-      }
-    }
-
-    double intensity_max, intensity_min, rt_min;
-    if (intensity_left >= intensity_right)
-    {
-      intensity_max = intensity_left;
-      intensity_min = intensity_right;
-      rt_min = best_right;
-    }
-    else 
-    {
-      intensity_max = intensity_right;
-      intensity_min = intensity_left;
-      rt_min = best_left;
-    }
-    // calculate the average noise level using the sin/cos rule
-    double delta_int = intensity_max - intensity_min;
-    double delta_rt = best_right - best_left;
-    double delta_rt_apex = std::fabs(rt_min-rt_apex);
-    double delta_int_apex = delta_int*delta_rt_apex/delta_rt;
-
-    avg_noise_level = intensity_min + delta_int_apex;
-    //NOTE: formula for calculating the background using the trapezoidal rule (future PR)
-    // background = intensity_min*delta_rt + 0.5*delta_int*delta_rt;
-
-    // calculate the background
-    background = 0.0;
-    for (MSChromatogram::const_iterator it = std::next(chromatogram.begin()); it != chromatogram.end(); ++it)
-    {
-      MSChromatogram::const_iterator it_prev = it;
-      --it_prev; //previous point
-
-      if (it->getMZ() >= best_left && it_prev->getMZ() < best_right)
-      {
-        // calculate the background using the formula
-        // y = mx + b where x = retention time, m = slope, b = left intensity
-        double delta_int = intensity_right - intensity_left; // sign will determine line direction
-        double delta_rt_current = (it->getMZ() - best_left);
-        double background_int_current = delta_int/delta_rt*delta_rt_current + intensity_left;
-        background = background + background_int_current;
+        max_height = it->getIntensity();
+        peak_apex_RT = it->getRT();
       }
     }
+    Param params;
```

Comment:
It would be best to have all params initialized during the constructor call to MRMTransitionGroupPicker and then passed to PeakIntegrator as needed.

Reply:
Depending on what we're trying to calculate, a different value for the same parameter is needed. Like in the case of the method `calculatePeakApexInt_()` which calculates (at least from what I understand through the code) two kinds of integrals. I could move the "baseline_type" setting to the constructor as you recommended, but "integration_type" would still need to change during code execution.

Reply:
Why would the "integration_type" change during code execution?  It should also be something that is specified by the user prior to executing the method.